.github/workflows/auto-upgrade-certified-connectors-cdk.yml

```diff
@@ -0,0 +1,230 @@
+name: Auto Upgrade CDK for Certified Connectors
+on:
+  schedule:
+    # Cron doesn't have a good way to specify "every X weeks" or "the first Monday of every month",
+    # so we'll just run every Monday at 16:00 UTC.
+    # The `limit-run-frequency` job has logic to limit this
+    # to only actually _do_ anything every on the first Monday of the month.
+    - cron: "0 16 * * 1"
+  workflow_dispatch:
+  workflow_call:
+
+jobs:
+  limit-run-frequency:
+    runs-on: ubuntu-latest
+    outputs:
+      is_valid_run: ${{ steps.check.outputs.is_valid_run }}
+    steps:
+      - id: check
+        # If the day-of-month is <= 7, then we're in the first week of the month.
+        # This workflow only runs on Mondays, so that would imply it's the first monday of the month.
+        run: |
+          if test $(date +%d) -le 7; then
+            echo "is_valid_run=true" >> "$GITHUB_OUTPUT"
+          else
+            echo "is_valid_run=false" >> "$GITHUB_OUTPUT"
+          fi
+  list-certified-connectors:
+    needs: limit-run-frequency
+    if: ${{ needs.limit-run-frequency.outputs.is_valid_run == 'true' }}
+    name: List Certified Connectors
+    runs-on: ubuntu-24.04
+    outputs:
+      connectors: ${{ steps.list-connectors.outputs.connectors }}
+    steps:
+      - name: Checkout Airbyte
+        uses: actions/checkout@08eba0b27e820071cde6df949e0beb9ba4906955 # v4
+
+      - name: List certified connectors
+        id: list-connectors
+        run: |
+          # datagen and dev-null aren't certified, but we should probably keep them on the latest CDK anyway
+          connectors=(destination-dev-null source-datagen)
+          for dir in airbyte-integrations/connectors/*; do
+            metadata_file="${dir}/metadata.yaml"
+            build_gradle="${dir}/build.gradle"
+            build_gradle_kts="${dir}/build.gradle.kts"
+
+            # If metadata.yaml exists and says we're certified
+            if test -f "$metadata_file" && grep -q "supportLevel: certified" "$metadata_file"; then
```

Comment:
should we use `yq` instead of `grep`? grep seems a little more likely to fail if there's a minor format change